packages/odyssey-react-mui/src/labs/TopNav.tsx

```diff
@@ -0,0 +1,430 @@
+/*!
+ * Copyright (c) 2022-present, Okta, Inc. and/or its affiliates. All rights reserved.
+ * The Okta software accompanied by this notice is provided pursuant to the Apache License, Version 2.0 (the "License.")
+ *
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0.
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *
+ * See the License for the specific language governing permissions and limitations under the License.
+ */
+
+import styled from "@emotion/styled";
+import { memo, useMemo, ReactElement, useCallback, KeyboardEvent } from "react";
+
+import { Box } from "../Box";
+import type { HtmlProps } from "../HtmlProps";
+import {
+  AuraIcon,
+  AuraWordmarkIcon,
+  QuestionCircleIcon,
+  SettingsIcon,
+} from "../icons.generated";
+import { Link } from "../Link";
+import {
+  DesignTokens,
+  useOdysseyDesignTokens,
+} from "../OdysseyDesignTokensContext";
+
+export type TopNavLinkItem = {
+  id: string;
+  label: string;
+  /**
+   * link added to the nav item. if it is undefined, static text will be displayed.
+   * fires onClick event when it is passed
+   */
+  href?: string;
+  /**
+   * determines whether the link item is diabled
+   */
+  isDisabled?: boolean;
+  /**
+   * Event fired when the nav item is clicked
+   */
+  onClick?(): void;
+  /**
+   * The link target prop. e.g., "_blank"
+   */
+  target?: string;
+};
+
+export type UserProfileProps = {
+  /**
+   * Logged in user profile icon to be displayed in the top nav
+   */
+  profileIcon?: ReactElement;
+  /**
+   * Logged in user info to be displayed in the top nav
+   */
+  userName: string;
+  /**
+   * Org name of the logged in user
+   */
+  orgName: string;
+};
+
+export type TopNavProps = {
+  /**
+   * Determines whether to display the logo (aura & wordmark)
+   */
+  hasLogo?: boolean;
+  /**
+   *  Determines whether the search box is displayed
+   */
+  search?: ReactElement;
+  /**
+   * Nav links in the top nav
+   */
+  topNavLinkItems: TopNavLinkItem[];
+  /**
+   * Displays an additional button after the link items if it is passed
+   */
+  additionalNavItem?: ReactElement;
```

Comment:
This is similar to the above `search` field that consuming apps can pass through, as one of the apps (Workflows) seems to have an extra button input in their UI.

Reply:
Same comment as above. Don't mind the inversion of control here but would prefer a prop name like `AdditionalNavItemComponent`. Just a note, if we do that, the description of the prop isn't quite right
`Displays an additional button after the link items if it is passed` isn't necessarily true. We can tell them that is what we'd prefer be passed in though

Reply:
yeah, I was having a hard time naming this prop :) I'll update the name/desc.